furnace.c

```diff
@@ -500,6 +506,61 @@ init_pilot(furnace_context_t *ctx)
   ctx->pilot.is_enabled = false;
 }
 
+static void
+init_stdio(furnace_context_t* ctx)
+{
+  ctx->stdio.parser = ctx->stdio.buffer;
+  memset(&ctx->stdio.buffer, 0, BUF_SIZE);
+}
+
+static void
+reset_stdio_data(furnace_context_t* ctx)
+{
+  memset(&ctx->stdio.buffer, 0, BUF_SIZE);
+  ctx->stdio.parser = ctx->stdio.buffer;
+}
+
+static void
+stdio_command_handler(furnace_context_t* ctx)
+{
+  void
+  send_stdio(const char* msg, const size_t msg_len)
+  {
+    printf(msg);
+  }
+
+  command_handler(ctx, ctx->stdio.buffer, &send_stdio);
+}
+
+void
+do_stdio_work(furnace_context_t* ctx)
+{
+  while(1) {
+    uint8_t c = getchar_timeout_us(1);
```

Comment:
We shouldn't probably block, even for 1 microsecond. If there is no character for us to consume, let's just return early.